pkg/ingester/ingester_v2.go

```diff
@@ -455,3 +488,93 @@ func (i *Ingester) closeAllTSDB() {
 	i.userStatesMtx.Unlock()
 	wg.Wait()
 }
+
+/* cleanup walks the TSDB directory for each user to determine if the TSDB should be closed.
+ * If
+ * - The directory is empty of blocks
+ * OR
+ * - The directory has every block marked as shipped
+ * AND
+ * - The TSDB hasn't been written to within the alloted time period
+ * THEN
+ * - Close the TSDB, stop the shipper routine, and delete the directory and it's contents.
+ */
+func (i *Ingester) cleanup() error {
+	ts := time.Now()
+	defer func() {
+		i.metrics.cleanupDuration.Observe(time.Since(ts).Seconds())
+	}()
+
+	userIDs := i.getUsers()
+
+	unshipped, err := unshippedBlocks(i.cfg.TSDBConfig.Dir)
+	if err != nil {
+		return err
+	}
+
+	wg := &sync.WaitGroup{}
+	for _, user := range userIDs {
+
+		// Only try and process users that don't have unshipped blocks
+		if len(unshipped[user]) != 0 {
+			continue
+		}
+
+		// Get the last written time stamp
+		udb := i.getTSDB(user)
+		if udb == nil {
+			level.Warn(util.Logger).Log("msg", "unable to find user tsdb during cleanup", "user", user)
+			continue
+		}
+
+		udb.Lock()
+		last := udb.lastWritten
+		udb.Unlock()
+
+		// If the TSDB is inactive, close and remove it.
+		if ts.Sub(last) > i.cfg.TSDBConfig.MaxStaleAge {
```

Comment:
We could reverse with a guard, like you did above for other checks:

```
if ts.Sub(last) <= i.cfg.TSDBConfig.MaxStaleAge {
  continue
}
```

Reply:
Sure. Changed.